spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/elasticsearch/ElasticsearchRestClientConfigurations.java

```diff
@@ -114,25 +123,48 @@ RestClient elasticsearchRestClient(RestClientBuilder builder) {
 
 		private final ElasticsearchRestClientProperties properties;
 
+		private CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
```

Comment:
Creating a field for the `CredentialsProvider` and allowing it to be injected doesn't seem to be related to supported user info in the URI. Can you please revert this part of the changes?

Reply:
I added this to make testing easer, because we can not access `credentialsProvider` field from `InternalHttpAsyncClient`. It can be also useful in some cases when the user needs to use a custom `CredentialsProvider`.

Reply:
I don't think it's safe to expose a `CredentialProvider` as a bean. In this case the credentials that it provides are specific to Elasticsearch, but there's nothing about the bean that indicates that's the case. It could be accidentally consumed and used anywhere that's working with an Apache HTTP client, potentially leaking credentials to another service.

For this sort of situation, we prefer to use AssertJ's support for extracting fields. In this case it would look something like this:

```java
assertThat(client).extracting("client")
		.extracting("credentialsProvider", InstanceOfAssertFactories.type(CredentialsProvider.class))
		.satisfies((credentialsProvider) -> {
			Credentials uriCredentials = credentialsProvider.getCredentials(new AuthScope("localhost", 9200));
			assertThat(uriCredentials.getUserPrincipal().getName()).isEqualTo("user");
			assertThat(uriCredentials.getPassword()).isEqualTo("password");
			Credentials defaultCredentials = credentialsProvider.getCredentials(new AuthScope("localhost", 9201));
			assertThat(defaultCredentials.getUserPrincipal().getName()).isEqualTo("admin");
			assertThat(defaultCredentials.getPassword()).isEqualTo("admin");
		});
```

Reply:
Thank you for the AssertJ example, @wilkinsona 

I will update tests.

Reply:
Fixed.